hack/licence-detector/README.md

```diff
@@ -0,0 +1,76 @@
+Licence Detector
+================
+
+This directory contains the scripts to generate licence notice and dependency information documentation.
+
+- `generate-notice.sh`: Invoked by `make generate-notice-file` to automatically generate `NOTICE.txt` and `docs/dependencies.asciidoc`.
+- `generate-image-deps.sh`: Manually invoked script to update the container image dependency information in `docs/container-image-dependencies.csv`.
```

Comment:
Can you clarify what the CSV is intended for? It doesn't look like it's used as an input to anything else. I was also a little surprised when I was testing it out and it needed to sudo

Reply:
The CSV is `include`d by `docs/dependencies.asciidoc` and contains the licence information for the Docker image. The sudo request by Tern is indeed a bit worrying but apparently it's needed to mount the procfs file system:

> Tern requires root privileges to run because it needs to mount procfs in order to run commands within a chroot environment and call the Docker CLI. It is enough if you have configured sudo; Tern will ask for your password before running any priviledged commands.

I wish they supported OCI images rather than relying so heavily on Docker daemon and CLI but it's the only reliable tool I have managed to find. The alternative is to [manually maintain a list of dependencies in the Docker image](https://github.com/elastic/cloud/blob/f418f0ff778977078b827669c629c22983924a42/found-deployment/cloud-enterprise/baseimages/enterprise-base/Dockerfile#L15)

Reply:
Got it, thanks for the clarification. I wonder if there's a clearer phrase than "container image dependency" since our direct dependencies are also included in the image (just inside the binary). Maybe changing to "container base image dependency..."?